libs/types/src/tokens.rs

```diff
@@ -79,6 +80,29 @@ impl From<StakingCurrency> for CurrencyId {
 	}
 }
 
+impl<Prefix> GeneralCurrencyIndex<Prefix> for CurrencyId
+where
+	Prefix: Get<[u8; 12]>,
+{
+	type CurrencyId = CurrencyId;
+	type GeneralIndex = u128;
+
+	fn get_general_index(
```

Comment:
I would implement this as a `Into` / `From` `GeneralCurrencyIndex` to/from `CurrencyId` instead.

Reply:
Could you give a short feedback if [c064dca](https://github.com/centrifuge/centrifuge-chain/pull/1292/commits/c064dca5a9c793210246fcc47e65b63636f22b63) matches your expectations? Then I will add the rest.